src/Altinn.Apps/AppTemplates/AspNet/App.IntegrationTests/Data/apps/tdd/complex-process/logic/Validation/ValidationHandler.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Threading.Tasks;
+using Altinn.App.Services.Interface;
+using Altinn.Platform.Storage.Interface.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+// using Altinn.App.Models; // Uncomment this line to refer to app model(s)
+
+
+namespace App.IntegrationTests.Mocks.Apps.tdd.complex_process.AppLogic.Validation
+{
+    public class ValidationHandler
+    {
+        private IHttpContextAccessor _httpContextAccessor;
+        private IInstance _instanceService;
+        public ValidationHandler(IInstance instanceService, IHttpContextAccessor httpContextAccessor = null)
+        {
+            _httpContextAccessor = httpContextAccessor;
+            _instanceService = instanceService;
+        }
+
+        /// <summary>
+        /// Handles all custom validations that are not covered by the data model validation.
+        /// </summary>
+        /// <remarks>
+        /// Validations that fail should be handled by updating the validation result object,
+        /// see example.
+        /// </remarks>
+        /// <param name="validationResults">Object to contain any validation results</param>
+        /// <example>
+        /// if (instance.GetType() == typeof([model class name])
+        /// {
+        ///     // Explicitly cast instance to correct model type
+        ///     [model class name] model = ([model class name])object;
+        ///
+        ///     // Perform validations
+        ///     if ([some condition])
+        ///     {
+        ///         validationResults.Add(new ValidationResult([error message], new List<string>() {[affected field id]} ));
+        ///     }
+        /// }
+        /// </example>
+        public async Task Validate(object instance, ModelStateDictionary validationResults)
+        {
+            /*   HttpContext context = _httpContextAccessor.HttpContext;
```

Comment:
Mest for diskusjon @nkylstad @acn-sbuad  har Instance i dette tilfellet informasjon om state? Slik ta man kan validere basert på hvilket steg man er i?

Reply:
VI har snakket litt om akkurat denne valideringen. At metoden her først og fremst går på datavalidering. Vi har derfor tenkt tanken at man kunne laget en ny task som heler "ValidateTask" som trigges på instansnivå der man kunne lagt inn sjekker av typen: er innsender over 20 år, er klokken over 18.. Valideringsmetoden vi har nå trigges kun dersom det er dataelementer knyttet til current task.

Reply:
Høres fornuftig ut.